File: apps/tasks-service/src/main/java/com/pactomais/tasksservice/task/Task.java
package com.pactomais.tasksservice.task;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UuidGenerator;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @UuidGenerator
    private UUID id;

    @Column(nullable = false)
    private String titulo;

    @Column(columnDefinition = "text")
    private String descricao;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskPriority prioridade;

    @Column(name = "responsavel_id")
    private UUID responsavelId;

    @Column(name = "responsavel_nome")
    private String responsavelNome;

    @Column(name = "time_id", nullable = false)
    private UUID timeId;

    @Column(name = "time_nome", nullable = false)
    private String timeNome;

    @CreationTimestamp
    @Column(name = "data_criacao", nullable = false, updatable = false)
    private Instant dataCriacao;

    private LocalDate prazo;
}

File: apps/web/src/types/task.ts
export type TaskStatus = "PENDENTE" | "EM_ANDAMENTO" | "CONCLUIDA"

export type TaskPriority = "BAIXA" | "MEDIA" | "ALTA"

export interface Task {
  id: string
  titulo: string
  descricao: string | null
  status: TaskStatus
  prioridade: TaskPriority
  responsavelId: string | null
  responsavelNome: string | null
  timeId: string
  timeNome: string
  dataCriacao: string
  prazo: string | null
}

export interface Page<T> {
  content: T[]
  totalElements: number
  totalPages: number
  number: number
  size: number
}

export interface TaskFilters {
  status?: TaskStatus
  responsavelId?: string
  prioridade?: TaskPriority
  page?: number
}
